modulemd/tests/test-modulemd-modulestream.c

```diff
@@ -331,6 +332,136 @@ module_stream_v2_test_documentation (ModuleStreamFixture *fixture,
   g_clear_object (&stream);
 }
 
+static void
+module_stream_v1_test_tracker (ModuleStreamFixture *fixture,
+                               gconstpointer user_data)
+{
+  g_autoptr (ModulemdModuleStreamV1) stream = NULL;
+  g_autofree gchar *tracker_prop = NULL;
+  const gchar *tracker = NULL;
+
+  stream = modulemd_module_stream_v1_new (NULL, NULL);
+
+  // Check the defaults
+  g_object_get (stream, MMD_TEST_TRACKER_PROP, &tracker_prop, NULL);
+  tracker = modulemd_module_stream_v1_get_tracker (stream);
+
+  g_assert_null (tracker);
+  g_assert_null (tracker_prop);
+
+  g_clear_pointer (&tracker_prop, g_free);
+
+  // Test property setting
+  g_object_set (stream, MMD_TEST_TRACKER_PROP, MMD_TEST_DOC_TEXT, NULL);
+
+  g_object_get (stream, MMD_TEST_TRACKER_PROP, &tracker_prop, NULL);
+  tracker = modulemd_module_stream_v1_get_tracker (stream);
+
+  g_assert_cmpstr (tracker, ==, MMD_TEST_DOC_TEXT);
+  g_assert_cmpstr (tracker_prop, ==, MMD_TEST_DOC_TEXT);
+
+  g_clear_pointer (&tracker_prop, g_free);
+
+  // Test set_tracker
+  modulemd_module_stream_v1_set_tracker (stream, MMD_TEST_DOC_TEXT2);
+
+  g_object_get (stream, MMD_TEST_TRACKER_PROP, &tracker_prop, NULL);
+  tracker = modulemd_module_stream_v1_get_tracker (stream);
+
+  g_assert_cmpstr (tracker, ==, MMD_TEST_DOC_TEXT2);
+  g_assert_cmpstr (tracker_prop, ==, MMD_TEST_DOC_TEXT2);
+
+  g_clear_pointer (&tracker_prop, g_free);
+
+  // Test setting it to NULL
+  g_object_set (stream, MMD_TEST_TRACKER_PROP, NULL, NULL);
+
+  g_object_get (stream, MMD_TEST_TRACKER_PROP, &tracker_prop, NULL);
+  tracker = modulemd_module_stream_v1_get_tracker (stream);
+
+  g_assert_null (tracker);
+  g_assert_null (tracker_prop);
+
+  g_clear_pointer (&tracker_prop, g_free);
+
+  // Test Unicode values
```

Comment:
According to the docs, the `tracker` is:
`The website address of the upstream bug tracker for this module.`

So it doesn't really make sense to have Unicode characters in a website URL. Now we should either remove this test assuming no one is ever going to add a URL containing Unicode or handle this case in the main function by returning `NULL` whenever we encounter an invalid URL.

@sgallagher What do you suggest?

Reply:
Leave it as-is for now. There's nothing *stopping* someone right now from putting unicode into the metadata here. We should probably look at using http-parser or llhttp to validate that we have validly-formatted URLs though. Mind opening a ticket to track that?